src/gt4py/next/program_processors/runners/dace/gtir_to_sdfg_types.py

```diff
@@ -152,23 +152,30 @@ def get_symbol_mapping(
         assert isinstance(outer_desc, dace.data.Array)
         # origin and size of the local dimension, in case of a field with `ListType` data,
         # are assumed to be compiled-time values (not symbolic), therefore the start and
-        # stop range symbols of the inner field only extend over the global dimensions
-        return (
-            {
-                gtx_dace_utils.range_start_symbol(dataname, dim): (self.origin[i])
-                for i, dim in enumerate(self.gt_type.dims)
-            }
-            | {
-                gtx_dace_utils.range_stop_symbol(dataname, dim): (
-                    self.origin[i] + outer_desc.shape[i]
-                )
-                for i, dim in enumerate(self.gt_type.dims)
+        # stop range symbols of the inner field only extend over the global dimensions.
+        if isinstance(self.gt_type.dtype, ts.ListType):
+            local_dim = self.gt_type.dtype.offset_type
+            assert local_dim is not None
+            full_dims = gtx_common.order_dimensions([*self.gt_type.dims, local_dim])
+            globals_size = [
+                size
+                for dim, size in zip(full_dims, outer_desc.shape, strict=True)
+                if dim != local_dim
+            ]
+        else:
+            globals_size = outer_desc.shape
+
+        symbol_mapping: dict[str, dace.symbolic.SymbolicType] = {}
+        for dim, origin, size in zip(self.gt_type.dims, self.origin, globals_size, strict=True):
```

Comment:
```suggestion
        for dim, origin, size in zip(gtx_common.order_dimensions(self.gt_type.dims), self.origin, globals_size, strict=True):
```
Should this be like this? I guess there are no local dimensions in `self.gt_type.dims` there but still

Reply:
Discussed offline. `self.gt_type.dims` should already be ordered